ci-operator/step-registry/openshift-extended/test/openshift-extended-test-commands.sh

```diff
@@ -35,6 +35,17 @@ then
     export GUEST_KUBECONFIG=${SHARED_DIR}/nested_kubeconfig
 fi
 
+# restore external oidc cache dir for oc
+if [[ -r "$SHARED_DIR/oc-oidc-token" ]] && [[ -r "$SHARED_DIR/oc-oidc-token-filename" ]]; then
+    echo "Restoring external OIDC cache dir for oc"
+    export KUBECACHEDIR
+    KUBECACHEDIR="/tmp/output"
+    token_cache_dir="$KUBECACHEDIR/oc"
+    mkdir -p "$token_cache_dir"
+    cat "$SHARED_DIR/oc-oidc-token" > "$token_cache_dir/$(cat "$SHARED_DIR/oc-oidc-token-filename")"
```

Comment:
@fxierh I still can not understand what it means after this.
does it means KUBECONFIG file does not take effective and replaced with it?

Reply:
@kuiwang02 

After logging in (with `openshift-extended-web-tests-ext-oidc-cli-login-commands.sh`), the ${SHARED_DIR}/KUBECONFIG of our external OIDC cluster is altered to something like
```bash
...
users:
- name: clusters-hypershift-ci-279587-admin
  user:
    client-certificate-data: ...
    client-key-data: ...
- name: oidc-user-test:hypershift-qe-ci@redhat.com/a44edf9442fb34f0e865d14ef74f4e05-7cc7fba2973e24e3-elb-us-east-2-amazonaws-com:6443
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1
      args:
      - get-token
      - --issuer-url=https://login.microsoftonline.com/.../v2.0
      - --client-id=...
      - --callback-address=127.0.0.1:8080
      - --extra-scopes=email,offline_access
      command: oc
      env: null
      installHint: Please be sure that oc is defined in $PATH to be executed as credentials
        exec plugin
      interactiveMode: IfAvailable
      provideClusterInfo: false
```

The following happens when an `oc` command (e.g. in `openshift-extended-test-commands.sh`) in invoked against the KUBECONFIG above:
1. `oc get-token ...` (as in the KUBECONFIG above) in invoked. This command will initiate an interactive login process unless it finds an already available token under `"$token_cache_dir"`, which it should be able to, because the token (generated in the previous `openshift-extended-web-tests-ext-oidc-cli-login` step) is restored to $token_cache_dir" by:
```bash
cat "$SHARED_DIR/oc-oidc-token" > "$token_cache_dir/$(cat "$SHARED_DIR/oc-oidc-token-filename")"
```
2. The token retrieved by `oc get-token ...` is passed to the API server for validation. If everything goes well, the API server will respond back with the data we requested for.

Reply:
In brief, `${SHARED_DIR}/KUBECONFIG` is still effective. It's just formulated in a special way that suggests the `oc` command to invoke another command (`oc get-token ...` in our case) to retrieve an OIDC token and send it to the API server for validation.

Reply:
got it, thanks